File: update_nav.js
const fs = require('fs');
const path = require('path');

function processHtmlFiles(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            if (file !== 'assets' && file !== '.gemini') {
                processHtmlFiles(fullPath);
            }
        } else if (file.endsWith('.html')) {
            let content = fs.readFileSync(fullPath, 'utf8');
            // Remove the PPE link completely. It looks generally like:
            // <li><a href="ppe.html">Учёт спецодежды</a></li>
            // or <li><a href="../products/ppe.html">Учёт спецодежды</a></li>
            // or <li><a href="products/ppe.html">Учёт спецодежды</a></li>
            const ppeRegex = /[ \t]*<li><a href="[^"]*ppe\.html">Учёт спецодежды<\/a><\/li>\s*/g;
            content = content.replace(ppeRegex, '\n');
            
            // Replace the text of the RFID link.
            const rfidRegex = /(<li><a href="[^"]*rfid\.html">)RFID-решения(<\/a><\/li>)/g;
            content = content.replace(rfidRegex, '$1RFID и спецодежда$2');
            
            fs.writeFileSync(fullPath, content);
            console.log(`Updated nav in ${fullPath}`);
        }
    }
}

const targetDir = process.cwd();
processHtmlFiles(targetDir);
console.log('Navigation update complete.');

File: update_nav.py
import os
import re

directory = '.'

for root, dirs, files in os.walk(directory):
    if 'assets' in root or '.gemini' in root:
        continue
    for file in files:
        if file.endswith('.html') and file != 'ppe.html':
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Remove ppe.html link
            content = re.sub(r'[ \t]*<li><a href="[^"]*ppe\.html">Учёт спецодежды</a></li>[ \t]*\n?', '', content)
            
            # Rename rfid link
            content = re.sub(r'(<li><a href="[^"]*rfid\.html">)RFID-решения(</a></li>)', r'\1RFID и спецодежда\2', content)
            
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(content)
            
            print(f"Updated {filepath}")
print("Done")
